core/src/Finalizer.cpp

```diff
@@ -0,0 +1,61 @@
+/*!
+ * @file Finalizer.cpp
+ *
+ * @date Sep 11, 2024
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#include "include/Finalizer.hpp"
+
+namespace Nextsim {
+
+void Finalizer::registerFunc(const FinalFn& fn)
+{
+    auto& fns = functions();
+    fns.push_back(fn);
+}
+
+bool Finalizer::registerUnique(const FinalFn& fn)
+{
+    if (contains(fn))
+        return false;
+    registerFunc(fn);
+    return true;
+}
```

Comment:
The return value indicates whether the function was successfully registered (`true`) or not (`false`). I this is useful enough functionality, given the minimal cost. Even if it is not currently used.

Reply:
I guess my concern was, if you thought you were using it but you were not. Or if it was left in for some use case that we no longer need.

Do you think it would be more useful to raise an exception if the user tries to register an already registered function? Given the use in the rest of the code, my concern is that new developers are not likely to test the boolean return value.

Reply:
It's there as a service if the developer wants to know if the function was already registered. How it is used currently, I don't care, so I don't check. If a future developer does care, then they can check.